waku/node/node_factory.nim

```diff
@@ -0,0 +1,374 @@
+import
+  std/[options, sequtils],
+  chronicles,
+  chronos,
+  libp2p/peerid,
+  libp2p/protocols/pubsub/gossipsub,
+  libp2p/nameresolving/dnsresolver,
+  libp2p/crypto/crypto
+
+import
+  ./waku_node,
+  ./config,
+  ./builder,
+  ./peer_manager,
+  ./peer_manager/peer_store/waku_peer_storage,
+  ./peer_manager/peer_store/migrations as peer_store_sqlite_migrations,
+  ../waku_enr/sharding,
+  ../waku_core,
+  ../waku_rln_relay,
+  ../waku_dnsdisc,
+  ../waku_archive,
+  ../waku_store,
+  ../waku_filter,
+  ../waku_filter_v2,
+  ../waku_peer_exchange,
+  ../waku_lightpush/common,
+  ../waku_archive/driver/builder,
+  ../waku_archive/retention_policy/builder,
+  ../common/utils/parse_size_units,
+  ../../apps/wakunode2/external_config,
+  ../../apps/wakunode2/internal_config,
+  ../../apps/wakunode2/wakunode2_validator_signed
```

Comment:
I have mixed feelings regarding importing code from the `apps` directory.
On one hand, the node's configuration parameters are defined there and we need them in order to create a node.

On the other, code related to the app should be separate from the the waku node modules.

Not sure if we should move the configs out of the `app` directory or what would be the most appropriate way to go

Reply:
Good point! What if we move the `external_config`, `internal_config`, and `wakunode2_validator_signed` modules into the folder `waky/node/`?

Reply:
Sounds good :))

Reply:
So took a look and was able to move `internal_config.nim` and `wakunode2_validator_signed` but moving `external_config.nim` caused some conflicts because of its code related to parsing CLI commands, which also makes sense that is part of the `apps` directory.

Ideally we should decouple a node's configuration options to the application's CLI parameters.
The decoupling would make sense only if we find a way to do it without duplicating everything. So I left `external_config.nim` in `apps` until we find a better solution.